src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -537,23 +537,66 @@ private BoundBadExpression MissingDeconstruct(BoundExpression receiver, CSharpSy
 
-        internal BoundLocalDeconstructionDeclaration BindDeconstructionDeclaration(CSharpSyntaxNode node, VariableDeclarationSyntax declaration, DiagnosticBag diagnostics)
+        internal BoundLocalDeconstructionDeclaration BindDeconstructionDeclarationStatement(DeconstructionDeclarationStatementSyntax node, DiagnosticBag diagnostics)
         {
-            Debug.Assert(node.Kind() == SyntaxKind.LocalDeclarationStatement || node.Kind() == SyntaxKind.VariableDeclaration);
-            Debug.Assert(declaration.IsDeconstructionDeclaration);
-
-            var value = declaration.Deconstruction.Value;
-            return new BoundLocalDeconstructionDeclaration(node, BindDeconstructionDeclaration(node, declaration, value, diagnostics));
+            return new BoundLocalDeconstructionDeclaration(node, BindDeconstructionDeclaration(node, node.Assignment.VariableComponent, node.Assignment.Value, diagnostics));
         }
 
-        internal BoundDeconstructionAssignmentOperator BindDeconstructionDeclaration(CSharpSyntaxNode node, VariableDeclarationSyntax declaration, ExpressionSyntax right, DiagnosticBag diagnostics, BoundDeconstructValuePlaceholder rightPlaceholder = null)
+        internal BoundDeconstructionAssignmentOperator BindDeconstructionDeclaration(CSharpSyntaxNode node, VariableComponentSyntax declaration, ExpressionSyntax right, DiagnosticBag diagnostics, BoundDeconstructValuePlaceholder rightPlaceholder = null)
         {
-            ArrayBuilder<DeconstructionVariable> locals = BindDeconstructionDeclarationLocals(declaration, declaration.Type, diagnostics);
+            DeconstructionVariable locals = BindDeconstructionDeclarationLocals(declaration, diagnostics);
+            Debug.Assert(locals.HasNestedVariables);
+            var result = BindDeconstructionAssignment(node, right, locals.NestedVariables, diagnostics, isDeclaration: true, rhsPlaceholder: rightPlaceholder);
+            FreeDeconstructionVariables(locals.NestedVariables);
+            return result;
+        }
 
-            var result = BindDeconstructionAssignment(node, right, locals, diagnostics, isDeclaration: true, rhsPlaceholder: rightPlaceholder);
-            FreeDeconstructionVariables(locals);
+        private DeconstructionVariable BindDeconstructionDeclarationLocals(VariableComponentSyntax node, DiagnosticBag diagnostics)
+        {
+            switch (node.Kind())
+            {
+                case SyntaxKind.TypedVariableComponent:
+                    {
+                        var component = (TypedVariableComponentSyntax)node;
+                        return BindDeconstructionDeclarationLocals(component.Type, component.Designation, diagnostics);
+                    }
+                case SyntaxKind.TupleDeconstructionVariableComponent:
+                    {
+                        var component = (TupleDeconstructionVariableComponentSyntax)node;
+                        var builder = ArrayBuilder<DeconstructionVariable>.GetInstance();
```

Comment:
Sure.

---
In reply to: [73610829](https://github.com/dotnet/roslyn/pull/12917#discussion_r73610829) [](ancestors = 73610829)